src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -180,85 +67,62 @@ module Read =
     let inline len (bytes: EventBody) = bytes.Length
     let private resolvedEventLen (x: ITimelineEvent<EventBody>) = len x.Data + len x.Meta
     let private resolvedEventBytes events = events |> Array.sumBy resolvedEventLen
-    let private loggedReadSlice reader streamName batchSize requiresLeader startPos batchIndex (log: ILogger) ct: Task<_> = task {
+    let private loggedReadSlice reader streamName batchSize requiresLeader startPos batchIndex ct: Task<_> = task {
         let act = Activity.Current
         let! t, slice = readSliceAsync reader streamName batchSize startPos requiresLeader |> Stopwatch.time ct
         let bytes, count = slice.Messages |> resolvedEventBytes, slice.Messages.Length
-        let reqMetric: Log.Measurement = { stream = streamName; interval = t; bytes = bytes; count = count}
-        let evt = Log.Slice reqMetric
         if act <> null then act.IncMetric(count, bytes).AddLastVersion(slice.LastVersion) |> ignore
-        let log = if not (log.IsEnabled Events.LogEventLevel.Debug) then log else log |> Log.propResolvedEvents "Json" slice.Messages
-        (log |> Log.prop "startPos" startPos |> Log.prop "bytes" bytes |> Log.event evt).Information("Mdb{action:l} count={count} version={version}",
-            "Read", count, slice.LastVersion)
         return slice }
 
-    let private readBatches (log: ILogger) batchSize (readSlice: int64 -> int -> ILogger -> CancellationToken -> Task<StreamEventsSlice>)
+    let private readBatches fold initial tryDecode batchSize (readSlice: int64 -> int -> CancellationToken -> Task<StreamEventsSlice>)
```

Comment:
initial -> origin?
also batchSize should be set, probably early on outside and not passed in here